src/_posts/2017-08-04-splittable-do-fn.md

```diff
@@ -0,0 +1,449 @@
+  OffsetRange getInitialRestriction(String filename) {
+    return new OffsetRange(0, new File(filename).getSize());
+  }
+}
+```
+
+
+This hypothetical `DoFn` reads records from a single Avro file. Notably missing
+is the code for expanding a filepattern: it no longer needs to be part of this
+`DoFn`! Instead, the SDK includes a
+[Match.filepatterns()](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/io/Match.java)
+transform for expanding a filepattern into a `PCollection` of filenames, and
+different file format IOs can reuse the same transform, reading the files with
+different `DoFn`'s.
+
+This example demonstrates the benefits of increased modularity allowed by SDF:
+`Match` supports continuous ingestion of new files in streaming pipelines using
+`.continuously()`, and this functionality becomes automatically available to
+various file format IOs, e.g. `TextIO.read().watchForNewFiles()` [uses `Match`
+under the
+hood)](https://github.com/apache/beam/blob/f7e8f886c91ea9d0b51e00331eeb4484e2f6e000/sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java#L480).
+
+## Current status
+
+Splittable `DoFn` is a major new API, and its delivery and widespread adoption
+involves a lot of work in different parts of the Apache Beam ecosystem.  Some of
+that work is already completed and provides direct benefit to users via new IO
+connectors, however, a large amount of work is in progress or planned.
+
+As of writing, SDF is available for use in the Beam Java Direct runner and
+Dataflow Streaming runner, and implementation is in progress in the Flink and
+Apex runners; see [capability
+matrix]({{ site.baseurl }}/documentation/runners/capability-matrix/) for
+the current status.
+
+Several SDF-based transforms and IO connectors available for Beam users at HEAD
+and will be included in Beam 2.2.0: `TextIO` and `AvroIO` finally provide
+continuous ingestion of files (one of the most frequently requested features)
+via `.watchForNewFiles()`, backed by the utility transforms
+`Match.filepatterns().continuously()` and the more general
+[`Watch.growthOf()`](https://github.com/apache/beam/blob/f7e8f886c91ea9d0b51e00331eeb4484e2f6e000/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Watch.java)
+which support several other "power user" use cases.
+
+IOs shipped with Beam SDK based on the Source API will be changed to use SDF, in
+order to enable more flexible use cases: this transition is [pioneered by
+TextIO](http://s.apache.org/textio-sdf) and involves temporarily [executing SDF
+via the Source API](http://s.apache.org/sdf-via-source) to support runners
+lacking the ability to run SDF directly.
+
+In addition to enabling new IOs, work on SDF has influenced our thinking about
+other parts of the Beam programming model:
+
+* By providing unification in the final remaining part of the Beam programming
+model that was not batch/streaming agnostic (the `Source` API), SDF lead us to
```

Comment:
Wondering if you could do away with the first part of this bullet "By providing..." and just start with "SDF led us to..." to simplify things? 
lead -> led